### PR TITLE
Ingredient and Recipe class

### PR DESCRIPTION
**Pull request by ncy-xing**
Updates:
- Ingredient class supports units and unit-less items (e.g., 2 eggs). You must specify a unit in the constuctor; specify "" or None for unitless items.
- Recipe class organizes ingredients by category. You must specify a category when adding an ingredient. You can retrieve ingredients by category. 
- constants module defines string values for category and units. You can easily iterate over categories after importing this module. 
See main method in Recipe module for a test example.

### CODE REVIEW

File: src/constants.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+FILLINGS = "filling"
+LEAVENERS = "leavener"
+TOPPINGS = "toppings"
+
+CUP = "cup"
+OUNCE = "oz"
+TSP = "tsp"
+TBSP = "tbsp"
+
+CATEGORIES = [DRY_BASES, WET_BASES, SWEETENERS, FLAVORINGS, SPICES, FILLINGS, LEAVENERS, TOPPINGS]
```
Comment (ygebregz) [inline src/constants.py:15]: while i think this is fine, I think python also has enum classes that we can define but this does the same thing so for what we need it is probably fine.

**Comment by ygebregz** [comment]
For my recipe manager class, i'm expecting a dictionary with the different components and the score calculated for them. Are you going to do that once you implement the questions part?

From my understanding, should I be expecting something like this. 
```
probability_dist = {
"dry_bases" : 0.20,
"sweetners: : 0.90, 
....
}
```
Here, sweetner being 0.9 hints at the emotion shown by the user is happy. Therefore, that tells me that I need to mulitply the random generated amount by 0.9 to decide the amount. Are we thinking of something along those lines?

**Review by ygebregz — APPROVED** [review]
Looks good to me other wise.

**Comment by ncy-xing** [comment]
> For my recipe manager class, i'm expecting a dictionary with the different components and the score calculated for them. Are you going to do that once you implement the questions part?
> 
> From my understanding, should I be expecting something like this.
> 
> ```
> probability_dist = {
> "dry_bases" : 0.20,
> "sweetners: : 0.90, 
> ....
> }
> ```
> 
> Here, sweetner being 0.9 hints at the emotion shown by the user is happy. Therefore, that tells me that I need to mulitply the random generated amount by 0.9 to decide the amount. Are we thinking of something along those lines?


Good question--yes, I will implement a dict like that with the questions part. I think the Recipe class itself should be blind to those probabilities and we create Recipe objects with the actual amounts only after applying those probabilities.

**Pull request merged**